File: prims.c
#include<stdio.h>
int minDistance(int distance[],int mst[]);
//Less than 10 vertices
    int n = 5;
int main() {

    int a[5][5] = { { 0, 2, 0, 6, 0 }, 
                        { 2, 0, 3, 8, 5 }, 
                        { 0, 3, 0, 0, 7 }, 
                        { 6, 8, 0, 0, 9 }, 
                        { 0, 5, 7, 9, 0 } }; 
    int parent[10];
    int mst[10] = {0};
    int distance[10];
    for(int i=0;i<n;i++)
        distance[i] = 9999;
    distance[0] = 0;
    parent[0] = -1;

    for(int count=0;count<n;count++) {
        int u = minDistance(distance,mst);
        printf("%d",u);
        mst[u] = 1;

        for(int i=0;i<n;i++) {
            if(a[u][i] && mst[i] == 0 && a[u][i] < distance[i])
                parent[i] = u;
                distance[i] = a[u][i];
        }
    }

    // for(int k=0;k<n;k++) {
    //     printf("%d",parent[k]);
    // }

    // printf("%d",parent[2]);

    }

    int minDistance(int distance[],int mst[]) {
        int min=9999;
        int min_index;
        for(int i=0;i<n;i++) {
            if(mst[i]==0 && distance[i]<min) {
                min = distance[i];
                min_index = i;
            }
        }
        return min_index;
    }

File: default/prims.py
# Input Adjacency Matrix [x]
# Generate intial Distance, MST and Parent Lists [x]
# Perform Prims Algorithm Operation and Generate Parent List
# Build Visualization

import sys

# GLOBAL LISTS
Dist = []
MST = []
Parent = []
adjacency_matrix = []


def generate_initial_lists(V):
    for i in range(V):
        Dist.append(sys.maxsize)
        MST.append(False)
        Parent.append(-1)
    Dist[0] = 0
    # print('[{} {} {}]'.format(Dist, MST, Parent))
    generate_mst_set(V)


def generate_mst_set(V):
    ticker = True
    while ticker:
        d1 = min(Dist)
        parent = Dist.index(d1)
        MST[parent] = True
        Dist[parent] = sys.maxsize
        for i in range(V):
            if adjacency_matrix[parent][i] != 0 and MST[i] != True:
                Parent[i] = parent
                Dist[i] = min(Dist[i], adjacency_matrix[parent][i])
        if False not in MST:
            ticker = False
    print(Parent)
    # print('[{} {} {}]'.format(Dist, MST, Parent))


def generate_adjacency_matrix():
    print(":: ADJACENCY LIST CREATION ::")
    # print("Enter number of vertices :")
    # V = int(input())

    # for i in range(V):
    #     adjacency_matrix.append([])
    #     for j in range(V):
    #         print('Enter weight for VERTEX :: {} -> {}'.format(i, j))
    #         adjacency_matrix[i].append(int(input()))
    V = 6
    global adjacency_matrix
    adjacency_matrix = [[0, 4, 6, 0, 0, 0],
                        [4, 0, 6, 3, 4, 0],
                        [6, 6, 0, 1, 0, 0],
                        [0, 3, 1, 0, 2, 3],
                        [0, 4, 0, 2, 0, 7],
                        [0, 0, 0, 3, 7, 0]
                        ]

    generate_initial_lists(V)


generate_adjacency_matrix()
